src/GPU/gpu_assembler.jl

```diff
@@ -0,0 +1,81 @@
+struct GPUAssemblerSparsityPattern{Tv,Ti,VEC_FLOAT<:AbstractVector{Tv},SPARSE_MAT<:AbstractSparseArray{Tv,Ti}} <: AbstractSparseAssembler
+    K::SPARSE_MAT
+    f::VEC_FLOAT
+end
+
+
+function start_assemble(K::AbstractSparseArray{Tv}, f::AbstractVector{Tv}) where {Tv}
```

Comment:
Note for later: We might want to move this into an extension and do the following.
```suggestion
function start_assemble(K::Union{CuSparseMatrixCSC,CuSparseMatrixCSR}, f::CuVector)
```

Reply:
Well, I am not sure whether adapting an object that has a `union` attribute would work or not, I have to give it a try first but If it's adaptable so no problem of course.